etl/src/main/java/org/pih/hivmigration/etl/sql/LabResultMigrator.groovy

```diff
@@ -0,0 +1,261 @@
+            FROM hivmigration_lab_results r
+                     JOIN hivmigration_patients p ON r.source_patient_id = p.source_patient_id
+                     JOIN hivmigration_encounters e ON r.source_encounter_id = e.source_encounter_id
+            WHERE r.test_type ='ppd';
+            
+            -- Rapid test
+            SET @test_rapid_construct = (SELECT concept_id FROM concept WHERE uuid='A19F5A83-E960-413D-B93B-9270C53580A2');
+            INSERT INTO obs(obs_id, person_id, concept_id, encounter_id, obs_datetime, location_id, creator, date_created, voided, uuid)
+            SELECT r.obs_id, p.person_id, @test_rapid_construct, e.encounter_id, r.obs_datetime, e.location_id, 1, now(), 0, r.uuid
+            FROM hivmigration_lab_results r
+                     JOIN hivmigration_patients p ON r.source_patient_id = p.source_patient_id
+                     JOIN hivmigration_encounters e ON r.source_encounter_id = e.source_encounter_id
+            WHERE r.test_type ='tr';
+            
+            --
+            -- Fill in the constructs
+            --
+            -- Now for each construct we create the expected members.
+            
+            SET @specimen_number = (SELECT concept_id FROM concept WHERE uuid='162086AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA');
+            
+            -- Viral Load
+            --
+            
+            -- Add specimen number
+            INSERT INTO obs(person_id, concept_id, encounter_id, obs_datetime, location_id, obs_group_id, value_text, creator, date_created, voided, uuid)
+            SELECT p.person_id, @specimen_number, e.encounter_id, r.obs_datetime, e.location_id, r.obs_id, r.sample_id, 1, now(), 0, uuid()
+            FROM hivmigration_lab_results r
+                     JOIN hivmigration_patients p ON r.source_patient_id = p.source_patient_id
+                     JOIN hivmigration_encounters e ON r.source_encounter_id = e.source_encounter_id
+            WHERE r.test_type = 'viral_load';
+            
+            -- Add numeric VL values
+            SET @hvl_value = (SELECT concept_id FROM concept WHERE uuid='3cd4a882-26fe-102b-80cb-0017a47871b2');
+            INSERT INTO obs(person_id, concept_id, encounter_id, obs_datetime, location_id, obs_group_id, value_numeric, creator, date_created, voided, uuid)
+            SELECT p.person_id, @hvl_value, e.encounter_id, r.obs_datetime, e.location_id, r.obs_id, r.value_numeric,1, now(), 0, uuid()
+            FROM hivmigration_lab_results r
+                     JOIN hivmigration_patients p ON r.source_patient_id = p.source_patient_id
+                     JOIN hivmigration_encounters e ON r.source_encounter_id = e.source_encounter_id
+            WHERE r.value_numeric IS NOT NULL AND r.test_type ='viral_load';
+            
+            -- Add undetectable VL values
+            SET @hvl_qualitative = (SELECT concept_id FROM concept WHERE uuid='1305AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA');
```

Comment:
One question I'd ask - and I'd like to see how we do this in Malawi @cioan / @mogoodrich / @dmdesimone / @lnball .  If a viral load result is detectable, and we store the numeric value as a viral load obs, do we also store the qualitative result as "DETECTABLE" in addition to that?  Or do we only capture this when the answer is NOT_DETECTED?

Reply:
In Malawi EMR, we do not capture "DETECTABLE":
https://github.com/PIH/openmrs-module-pihmalawi/blob/master/omod/src/main/webapp/resources/htmlforms/viral_load_test_results.xml#L151

https://bwenzi.pih-emr.org/openmrs/dictionary/concept.htm?conceptId=8628